quinn-proto/src/connection/mod.rs

```diff
@@ -3381,3 +3384,15 @@ impl SentFrames {
             && self.retransmits.is_empty(streams)
     }
 }
+
+/// Whether `packet` is a valid stateless reset for this connection
+fn is_stateless_reset(packet: &[u8], peer_params: &TransportParameters) -> bool {
+    if packet.len() < RESET_TOKEN_SIZE + 5 {
+        return false;
+    }
+    let token = match peer_params.stateless_reset_token {
```

Comment:
I think we had this discussion before, but I think I would prefer `&` in the scrutinee to `ref` in the pattern -- you had a good rebuttal last time but I don't remember what it was and I can't think of a good reason in this case.

Reply:
I prefer `ref` patterns because they unambiguously communicate that you are binding a reference to the place. When matching on a value whose type isn't immediately apparent to the reader, the "match ergonomics" style makes it difficult to tell whether your pattern variables are getting the values in the place or a reference to it.

Rewrote to use `as_deref` in the course of inlining, so this is moot regardless.